integration_tests/structs_25.py

```diff
@@ -0,0 +1,20 @@
+from lpython import i32, dataclass
+from numpy import empty
+
+@dataclass
+class Foo:
+     x: i32
+     y: i32
+
+def init(foos: Foo[:]) -> None:
+    foos[0] = Foo(5, 21)
+
+def main0() -> None:
+    foos: Foo[1] = empty(1, dtype=Foo)
```

Comment:
I would just support the `Array[]` syntax for arrays of structs, like this:
```suggestion
    foos: Array[Foo, 1] = empty(1, dtype=Foo)
```
Then it should be possible to implement this cleanly in `lpython.py` as well as LPython.